src/Sample.TimeApi/Data/OpenTelemetryCollectingDeviceRepository.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Threading.Tasks;
+using OpenTelemetry.Trace;
+using Sample.Common;
+
+namespace Sample.TimeApi.Data
+{
+    public class OpenTelemetryCollectingDeviceRepository<TDeviceRepository> : IDeviceRepository
+           where TDeviceRepository : IDeviceRepository
+    {
+        private readonly TDeviceRepository repository;
+        private Tracer tracer;
+
+        public OpenTelemetryCollectingDeviceRepository(TDeviceRepository repository, TracerFactoryBase tracerFactory)
+        {
+            this.tracer = tracerFactory.GetTracer("sql");
+            this.repository = repository;
+        }
+
+        public async Task<DateTime> GetTimeFromSqlAsync()
+        {
+            var span = this.tracer.StartSpan(nameof(GetTimeFromSqlAsync), SpanKind.Client);
+            try
+            {
+                FailGenerator.FailIfNeeded(1);
+
+                return await this.repository.GetTimeFromSqlAsync();
+            }
+            catch (Exception ex)
+            {
+                span.SetAttribute("error", true);
```

Comment:
@SergeyKanzhelev 
Was wondering if that is the right way to make the Span as failed. Moreover, for Jaeger I had to add attribute "error"=true in order to show up nicely in the UI.